x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStopTransformAction.java

```diff
@@ -242,6 +242,17 @@ protected void taskOperation(
         }
 
         if (ids.contains(transformTask.getTransformId())) {
+            if (request.isForce()) {
+                try {
+                    // If force==true, we skip all the additional steps and only deregister and complete persistent task via shutdown call.
+                    // This way we ensure that the persistent task is removed ASAP (as opposed to being removed in one of the listeners).
+                    transformTask.shutdown();
+                    listener.onResponse(new Response(true));
+                } catch (ElasticsearchException ex) {
+                    listener.onFailure(ex);
+                }
+                return;
```

Comment:
Is there danger in continuing to perform the rest of cleanup after shutting down the persistent task? If something fails after shutting down the persistent task wouldn't that still be ok?

Reply:
Ah ok, thanks for the feedback.